### PR TITLE
Fix networked character physics hit volumes

### PR DESCRIPTION
**Pull request by mbalfour-amzn**
## What does this PR do?

This fixes the hit volumes on networked characters. Previously, the character physics didn't support rotation of the character controller, so the attached hit volumes always stayed unrotated, regardless of the visual character's rotation. With this fix, the hit volumes rotate with the networked characters.

Changes:
* Changed the NetworkCharacterComponent to set the rotation on the physics character.
* Changed the CharacterController to store a copy of the rotation and correctly apply it to the shadow body, and added a unit test to trivially validate that this works.
* Changed the NetworkHitVolumes to destroy themselves when the physics character is destroyed, so that they'll recreate correctly when a new physics character is created. (This fixes a teleportation bug with the hit volumes where they weren't associated correctly after the physics character was disabled / enabled)
* Added an assert to MultiplayerSystemComponent. There's an occasional crash related to corrupt net binding components ending up in the gathered entities list; this assert will hopefully catch the problem a little bit sooner than the crash to help with debugging if/when it happens again.

## How was this PR tested?

Ran the unit test before and after the changes to verify that it failed before and succeeded after.
Ran MPS before and after the changes and verified that the hit volumes now rotate correctly.

Before:
https://user-images.githubusercontent.com/82224783/228595750-6dc48ead-b1c5-40c5-95a0-a784397b979b.mp4

After:
https://user-images.githubusercontent.com/82224783/228595620-c32efc14-f14d-4f3b-9668-d4eb7dd9b3c1.mp4

### CODE REVIEW

**Review by greerdv — APPROVED** [review]

**Review by AMZN-Olex — APPROVED** [review]

File: Gems/Multiplayer/Code/Source/Components/NetworkCharacterComponent.cpp [hunk 1]
```diff
@@ -220,20 +220,23 @@ namespace Multiplayer
...
+        if (GetParent().m_physicsCharacter != nullptr)
         {
-            return GetEntity()->GetTransform()->GetWorldTranslation();
+            GetParent().m_physicsCharacter->SetRotation(GetEntity()->GetTransform()->GetWorldRotationQuaternion());
+
+            if (velocity.GetLengthSq() > 0.0f)
+            {
+                GetParent().m_physicsCharacter->Move(velocity * deltaTime, deltaTime);
+                GetEntity()->GetTransform()->SetWorldTM(GetParent().m_physicsCharacter->GetTransform());
+                AZLOG(
```
Comment (AMZN-Gene) [inline Gems/Multiplayer/Code/Source/Components/NetworkCharacterComponent.cpp:231]: Log looks spammy, but surprised it’s always been there. I don’t recall seeing logs every time I TryMove. I guess leave it if it not a problem?

**Review by AMZN-Gene — APPROVED** [review]

File: Gems/Multiplayer/Code/Source/Components/NetworkCharacterComponent.cpp (hunk 1 above)
Comment (kberg-amzn) [reply Gems/Multiplayer/Code/Source/Components/NetworkCharacterComponent.cpp:231]: It's a tagged log, you have to enable it. We have a whole bloom filter sitting outfront of the new log system that allows you to turn on and off logs at runtime through the console. Try running:
Logger.EnableLog NET_Movement
Works way better than putting in debug logs, commenting them out, deleting them, needing to add them back in next time you run into a problem, etc..

**Review by kberg-amzn — APPROVED** [review]
Excellent

**Review by hultonha — APPROVED** [review]
Nice!

**Pull request merged**